### ISSUE TITLE
role=presentation not allowed in Docker version, but allowed in online version

### ISSUE THREAD

**Issue by VikoRifeo** [validator/validator#984]
I've been trying to automate my website publishing process. I'm using a static site generator, so that means rebuilding the site constantly and naïvely revalidating every page every time. I'm trying to move to my own instance of the validator for obvious reasons.

However, my private instance of the validator issues `Error: Bad value “presentation” for attribute “role” on element “img”` where the public validator accepts it (and where I can't see a problem at all). I've checked that the files being validated are the same whether they come from the automatic builder or manual. The automatically-built files pass the public validator, and the manually-built files fail the private validator.

The private validator is running in a Docker container running the image with ID `validator/validator@sha256:38a483195bfaeaed990afa4a7ba514b7eb7e9f5c21e532c893309dbef182249f`, which as of this writing is the current `latest` tag.

One of the affected tags, as an example, is this:

```
<IMG alt="Discord" role="presentation" src="/icons/discord.svg">
```

**Comment by sideshowbarker** [comment]
> However, my private instance of the validator issues `Error: Bad value “presentation” for attribute “role” on element “img”` where the public validator accepts it

Given the following document:

```html
<!DOCTYPE html>
<html lang="">
<title>Test</title>
<IMG alt="Discord" role="presentation" src="/icons/discord.svg">
```

… https://validator.w3.org/nu/ reports *Bad value `presentation` for attribute `role` on element `img`*:

https://validator.w3.org/nu/?showsource=yes&doc=data:text/html;charset=utf-8,%3C!DOCTYPE%20html%3E%3Chtml%20lang=%22%22%3E%3Ctitle%3ETest%3C/title%3E%3CIMG%20alt=%22Discord%22%20role=%22presentation%22%20src=%22/icons/discord.svg%22%3E

…and that’s expected and correct, because https://w3c.github.io/html-aria/#el-img says that when an `img` has a non-empty `alt` value, it can’t have `role=presentation`.

So if you’re not seeing that error message, maybe it’s because you previously set up some filtering to filter it out?

**Comment by VikoRifeo** [comment]
Hm. Well, my bad on the actual validity front, but I don't think I *can* have set up filters, as I'm using it from a shell script through curl.

**Comment by VikoRifeo** [comment]
Wait a second. The public instance I was using is at https://validator.nu, not https://validator.w3.org/nu. It links here for its source code, and I didn't even notice this was a W3 operation. Am I in the right place? validator.nu doesn't list its version—is it outdated? A fork? Do you even know?

**Comment by sideshowbarker** [comment]
Yeah https://validator.nu/ isn’t as up to date as https://validator.w3.org/nu

I don’t maintain https://validator.nu/ — I don’t have access to bring it up to date. But I do maintain https://validator.w3.org/nu and I keep it up to date with the very latest changes to the source.

So to make sure you’re getting the latest changes, always use https://validator.w3.org/nu

(The history is that https://validator.nu/ was the original instance of the checker, before the https://validator.w3.org/nu W3C instance was created. And for some time, the W3C instance still wasn’t the recommended checker to use, because it checked documents against the W3C fork of the HTML spec rather than the actual HTML spec. But https://validator.nu/ always checked against the actual HTML spec rather than the W3C fork. But that changed last year, and https://validator.w3.org/nu now checks only against the actual HTML spec.)
